broker-core/src/test/java/io/zeebe/broker/workflow/processor/DeploymentCreatedProcessorTest.java

```diff
@@ -110,6 +113,102 @@ public void shouldNotWriteCloseSubscriptionIfNotMessageStart() {
         .isEqualTo(MessageStartEventSubscriptionIntent.OPEN);
   }
 
+  @Test
+  public void shouldCloseSubscriptionWhenInCorrectOrder() {
+    // given
+    streamProcessor.start();
+
+    // when
+    writeMessageStartRecord(3, 1);
+    writeNoneStartRecord(7, 2);
+
+    // then
+    waitUntil(() -> rule.events().onlyMessageStartEventSubscriptionRecords().count() == 2);
+
+    final TypedRecord<MessageStartEventSubscriptionRecord> closeRecord =
+        rule.events()
+            .onlyMessageStartEventSubscriptionRecords()
+            .withIntent(MessageStartEventSubscriptionIntent.CLOSE)
+            .getFirst();
+
+    assertThat(closeRecord.getValue().getWorkflowKey()).isEqualTo(3);
+  }
+
+  @Test
+  public void shouldIgnoreOutdatedDeployment() {
+    // given
+    streamProcessor.start();
+
+    // when
+    streamProcessor.blockAfterMessageStartEventSubscriptionRecord(
+        r ->
+            r.getValue().getWorkflowKey() == 5
+                && r.getMetadata().getIntent() == MessageStartEventSubscriptionIntent.OPEN);
+    writeMessageStartRecord(5, 2);
+    waitUntil(() -> streamProcessor.isBlocked());
+
+    streamProcessor.blockAfterDeploymentEvent(
```

Comment:
Why is this blocking necessary?

Reply:
Also in the other test

Reply:
I think we don't want all the records in the test to be written at the immediately one after another. We want them to be written and processed, and only then write the next one. For instance, if we write version 1 and then 2 immediately one after the other then, when the CREATED record for version 1 is processed, it will already be outdated and it won't open or close any subscriptions.